pkg/controller/storagecluster/controller_test.go

```diff
@@ -5705,8 +5705,57 @@ func TestUpdateStorageClusterFeatureGates(t *testing.T) {
 	// is detected to be updated.
 	require.Equal(t, []string{oldPod.Name}, podControl.DeletePodName)
 
-	// TestCase: Change spec.featureGates
-	cluster.Spec.FeatureGates["feature1"] = "disabled"
+	// TestCase: Change spec.CSI.Enabled to false
+	cluster.Spec.CSI = &corev1.CSISpec{
+		Enabled: false,
+	}
+	k8sClient.Update(context.TODO(), cluster)
+
+	podControl.DeletePodName = nil
+
+	result, err = controller.Reconcile(context.TODO(), request)
+	require.NoError(t, err)
+	require.Empty(t, result)
+	require.Equal(t, []string{oldPod.Name}, podControl.DeletePodName)
+
+	// TestCase: Change spec.CSI back to true
+	cluster.Spec.CSI = &corev1.CSISpec{
+		Enabled: true,
+	}
+	k8sClient.Update(context.TODO(), cluster)
+
+	podControl.DeletePodName = nil
+
+	result, err = controller.Reconcile(context.TODO(), request)
+	require.NoError(t, err)
+	require.Empty(t, result)
+	require.Equal(t, []string{oldPod.Name}, podControl.DeletePodName)
+
+	// TestCase: Snapshot controller installed should not bounce pods
+	trueBool := true
+	cluster.Spec.CSI.InstallSnapshotController = &trueBool
+	k8sClient.Update(context.TODO(), cluster)
+
+	podControl.DeletePodName = nil
+
+	result, err = controller.Reconcile(context.TODO(), request)
+	require.NoError(t, err)
+	require.Empty(t, result)
+	require.Equal(t, []string{oldPod.Name}, podControl.DeletePodName)
```

Comment:
Ideally this should be nil. If the test is passing it means the pod is bouncing when it should not.

Reply:
Tried this case and yeah you're right. Found an issue in isCSIBounceRequired. Testing a few other cases now.

Reply:
I was missing `replaceOldPod` and a few other required things for these tests. Fixed this up and did some manually testing, looking good now.